File: sorted_squared_array/sorted_squared_array.py
# Complexity O(n), Space O(n)
def sorted_squared_array(arr):
    pos = len(arr) - 1
    res = [0] * len(arr)
    begin = 0
    end = len(arr) - 1
    while begin <= end:
        if abs(arr[begin]) < abs(arr[end]):
            res[pos] = pow(arr[end], 2)
            end -= 1
        else:
            res[pos] = pow(arr[begin], 2)
            begin += 1
        pos -= 1
    return res


if __name__ == '__main__':
    print(sorted_squared_array([-7, -5, -4, 3, 6, 8, 9]))

File: sorted_squared_array/sortedSquaredArray.go
package main

import (
	"fmt"
	"math"
)

func sortedSquaredArray(arr *[]int) []int {
	var begin, end, pos int
	n := len(*arr)
	res := make([]int, n)

	end = n - 1
	pos = end

	for begin <= end {
		if math.Abs(float64((*arr)[begin])) < math.Abs(float64((*arr)[end])) {
			res[pos] = int(math.Pow(float64((*arr)[end]), float64(2)))
			end -= 1
		} else {
			res[pos] = int(math.Pow(float64((*arr)[begin]), float64(2)))
			begin += 1
		}
		pos -= 1
	}
	return res
}

func main() {
	fmt.Println(sortedSquaredArray(&[]int{-7, -5, -4, 3, 6, 8, 9}))
}
